containerd-shim-v2/create.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2014,2015,2016 Docker, Inc.
+// Copyright (c) 2017 Intel Corporation
+// Copyright (c) 2018 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package containerdshim
+
+import (
+	"context"
+	"fmt"
+
+	vc "github.com/kata-containers/runtime/virtcontainers"
+	"github.com/kata-containers/runtime/virtcontainers/pkg/oci"
+
+	taskAPI "github.com/containerd/containerd/runtime/v2/task"
+
+	"github.com/kata-containers/runtime/pkg/katautils"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func create(ctx context.Context, s *service, r *taskAPI.CreateTaskRequest, netns string,
+	runtimeConfig *oci.RuntimeConfig) (*container, error) {
+
+	detach := !r.Terminal
+
+	// Checks the MUST and MUST NOT from OCI runtime specification
+	bundlePath, err := validBundle(r.ID, r.Bundle)
+	if err != nil {
+		return nil, err
+	}
+
+	ociSpec, err := oci.ParseConfigJSON(bundlePath)
+	if err != nil {
+		return nil, err
+	}
+
+	containerType, err := ociSpec.ContainerType()
+	if err != nil {
+		return nil, err
+	}
+
+	// Todo:
+	// Since there is a bug in kata for sharedPidNs, here to
+	// remove the pidns to disable the sharePidNs temporarily,
+	// once kata fixed this issue, we can remove this line.
+	removeNameSpace(&ociSpec, specs.PIDNamespace)
+
+	//set the network namespace path
+	//this set will be applied to sandbox's
+	//network config and has nothing to
+	//do with containers in the sandbox since
+	//networkNamespace has been ignored by
+	//kata-agent in sandbox.
+	for _, n := range ociSpec.Linux.Namespaces {
+		if n.Type != specs.NetworkNamespace {
+			continue
+		}
+
+		if n.Path == "" {
+			n.Path = netns
```

Comment:
The reason I had explained above for why replacing the network namespace with netns.

Since the "netns" is created by containerd's plugin, thus if containerd doesn't enabled any network plugin, then here the "netns" will be empty and yes it will be done later by CreateSandbox().